modules/deployment/src/test/resources/org/foo/withLifecycleListener/LeakedThread.java

```diff
@@ -6,26 +6,24 @@
  */
 package org.foo.withLifecycleListener;
 
+import java.util.concurrent.CountDownLatch;
+
 /**
  * Thread for testing purposes. The idea is to retain the Thread's Context ClassLoader until the Thread is terminated.
  */
 public class LeakedThread extends Thread {
 
-  private boolean stopRequested = false;
+  private final CountDownLatch latch = new CountDownLatch(1);
 
-  public synchronized void run() {
-    stopRequested = false;
-    while (!stopRequested) {
-      try {
-        wait();
-      } catch (InterruptedException e) {
-        // does nothing (keeps waiting)
-      }
+  public void run() {
```

Comment:
I think you need to add an endless loop In order to make the code the same than before when an InterruptedException is thrown.